src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptInlineRenameOperation.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeActions;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
+{
+#pragma warning disable RS0030 // Do not used banned APIs
+    /// <summary>
+    /// A <see cref="CodeActionOperation"/> for navigating to a specific position in a document and invoking inline rename.
+    /// When <see cref="CodeAction.GetOperationsAsync(System.Threading.CancellationToken)"/> is called an implementation
+    /// of <see cref="CodeAction"/> can return an instance of this operation along with the other 
+    /// operations they want to apply. For example, an implementation could generate a new <see cref="Document"/>
+    /// in one <see cref="CodeActionOperation"/> and then have the host editor navigate to that
+    /// <see cref="Document"/> and invoke rename at a given position using this operation.
+    /// </summary>
+#pragma warning restore RS0030 // Do not used banned APIs
+    internal class VSTypeScriptInlineRenameOperation(DocumentId documentId, int position)
+        : DocumentNavigationOperation(documentId, position)
+    {
```

Comment:
i would not do it this way.  i would just make a whole new operation just for the purpose of invoking the inline rename system.